controllers/cluster_controller.go

```diff
@@ -357,6 +383,54 @@ func (r *ClusterReconciler) postClusterProviderReconcile(ctx context.Context, lo
 	return controller.Result{}, nil
 }
 
+func (r *ClusterReconciler) updateStatus(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) error {
+	// An EKS-A Cluster managed by the CLI is deleted after the cluster finalizer is removed, leaving the CAPI cluster behind. In this case, we do not want to update
+	// the status. If we do, patching will throw a 404 NotFound error when trying to update the status of the already deleted
+	// Cluster object.
+	if !cluster.DeletionTimestamp.IsZero() && metav1.HasAnnotation(cluster.ObjectMeta, anywherev1.ManagedByCLIAnnotation) {
+		log.Info("Cluster managed by CLI has been deleted but CAPI cluster may remain, skipping updating cluster status")
+		return nil
+	}
+
+	// ObservedGeneration represents the last observed generation, and consequently represents that the current status is up to date.
+	// Then if observedGeneration is equal to generation AND the Cluster's "Ready" condition is "True", then we can skip another status update
+	if cluster.Status.ObservedGeneration == cluster.Generation && conditions.IsTrue(cluster, anywherev1.ReadyCondition) {
+		log.Info("Generation matches observedGeneration and the cluster is ready, skipping status update")
+		return nil
+	}
+
+	log.Info("Updating cluster status")
+	defer func() {
+		// Always update the readyCondition by summarizing the state of other conditions.
+		conditions.SetSummary(cluster,
+			conditions.WithConditions(
+				anywherev1.ControlPlaneReadyCondition,
+				anywherev1.WorkersReadyConditon,
+			),
+		)
+	}()
+
+	if err := r.updateControlPlaneStatus(ctx, log, cluster); err != nil {
+		return errors.Wrap(err, "updating controlplane status")
+	}
+
+	return nil
+}
+
+func (r *ClusterReconciler) updateControlPlaneStatus(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) error {
+	log.Info("Updating control plane status")
+	kcp, err := controller.GetKubeadmControlPlane(ctx, r.client, cluster)
+	if err != nil {
+		return errors.Wrapf(err, "getting kubeadmcontrolplane")
+	}
+
+	if err = clusters.UpdateControlPlaneInitializedCondition(cluster, kcp); err != nil {
+		return errors.Wrap(err, "updating control plane initialized condition")
+	}
+
+	return nil
+}
```

Comment:
what about just moving this whole thing to the clusters package
that way all the "status" logic is there and the controller just orchestrates when it's called

Reply:
Agreed. I've moved this to the clusters package